run.sh

```diff
@@ -1,33 +1,57 @@
 #!/bin/bash
 set -e
-cd `dirname $0`
-VENV_NAME=viam-venv
+
+UNAME=$(uname -s)
+
+VENV_NAME=".venv"
 echo "Current Directory: $(pwd)"
 echo "Checking for virtual environment folder..."
 
-if [ -d "viam-env" ]
-  then
-    echo "Virtual environment found, activating..."
-    source viam-env/bin/activate
-    echo "Virtual environment activated: $VENV_NAME"
+if [ -d "$VENV_NAME" ]; then
+  echo "Virtual environment found, activating..."
+  source "$VENV_NAME/bin/activate"
+
+  # Check if 'uv' can be imported in Python
+  if ! python3 -c "import uv" > /dev/null 2>&1; then
+    echo "'uv' not found in virtual environment. Recreating the virtual environment..."
+    deactivate
+    rm -rf "$VENV_NAME"
   else
-    echo "Setting up virtual environment..."
-    python3 -m venv --system-site-packages "$VENV_NAME"
-    source "$VENV_NAME/bin/activate"
-    echo "Virtual environment activated: $VENV_NAME"
-    echo "Installing dependencies from requirements.txt..."
+    echo "'uv' is installed, skipping recreation of virtual environment."
+  fi
+fi
 
-    while IFS= read -r requirement; do
-      if [ -n "$requirement" ]; then
-        echo "Installing $requirement..."
-        pip install "$requirement" || echo "Failed to install $requirement"
-      fi
-    done < requirements.txt
+# Create a new virtual environment if it doesn't exist or was removed
+if [ ! -d "$VENV_NAME" ]; then
+  echo "Setting up virtual environment..."
+  
+  if [ "$UNAME" = "Linux" ]; then
+    echo "Installing uv on Linux"
 
-    echo "Dependencies installation complete."
+    # Check if pip is installed
+    if ! command -v pip &> /dev/null; then
+      echo "'pip' not found. Installing pip..."
+      sudo apt-get install -y python3-pip
+    fi
 
+    pip install uv
+  fi
+  
+  if [ "$UNAME" = "Darwin" ]; then
+    echo "Installing uv on Darwin"
+    brew install uv
   fi
-# Be sure to use `exec` so that termination signals reach the python process,
-# or handle forwarding termination signals manually
-echo "Starting Python module..."
-exec python3 -m src $@
+  
+  uv venv --python=3.10
+  source "$VENV_NAME/bin/activate"
+  echo "Virtual environment activated: $VENV_NAME"
+  
+  echo "Installing dependencies from requirements.txt..."
+  uv pip install -r requirements.txt
+  echo "Dependencies installation complete."
+fi
+
+PYTHON_LIB_PATH=$(find .venv/lib -type d -name "python3.*" -print -quit)
```

Comment:
Does this depend on the host python version?

Reply:
do you mean does it depend on the specific python version used to create the virtual env on the host machine? in that case I think it would still be fine because it could be any version of python greater than 3.0. Let me know if I am answering the right question